lib/resty/http.lua

```diff
@@ -914,5 +957,99 @@ function _M.proxy_response(self, response, chunksize)
     until not chunk
 end
 
+function _M.set_proxy_options(self, opts)
+    self.proxy_opts = opts
```

Comment:
Since you modify `self.proxy_opts` in other code paths, please take `opts` by value so that user tables can be reused without being modified by the module.

e.g.

```lua
self.proxy_opts = tbl_copy(opts)  -- Take by value
```

Reply:
Changed.